cardea/problem_definition/prolonged_length_of_stay.py

```diff
@@ -0,0 +1,150 @@
+
+                        es = es.entity_from_dataframe(entity_id=self.target_entity,
+                                                      dataframe=update_es,
+                                                      index='identifier')
+
+                        return(es, self.target_entity,
+                               es[self.target_entity].df[self.target_label],
+                               cutoff_times)
+                    except ValueError:
+                        raise ValueError('Cutoff time label {} in table {} does not exist'
+                                         .format(self.cutoff_time_label, self.target_entity))
+
+        except ValueError:
+            updated_es = self.generate_target_label(es)
+            return self.generate_cutoff_times(updated_es)
+
+    def generate_target_label(self, es):
+        """Generates target labels in the case of having missing label in the entityset.
+
+            Args:
+            es: fhir entityset.
+            target_label: The target label of the prediction problem.
+            target_entity: The entity name in which it contains the target label.
+
+            Returns:
+            Updated entityset with the generated label.
+
+            Raises:
+            ValueError: An error occurs if the target label cannot be generated.
+            """
+        generate_from = 'Period'
+        start = 'start'
+        end = 'end'
+
+        try:
+            if(self.check_target_label(
+                es,
+                generate_from,
+                start) and
+               self.check_target_label(
+                    es,
+                    generate_from,
+                    end)):
+                try:
+                    self.check_target_label_values(
+                        es,
+                        generate_from,
+                        start)
+                    self.check_target_label_values(
+                        es,
+                        generate_from,
+                        end)
+                    es[generate_from].df[start] = pd.to_datetime(
+                        es[generate_from]
+                        .df[start])
+                    es[generate_from].df[end] = pd.to_datetime(
+                        es[generate_from].df[end])
+                    duration = (es[generate_from].df[end] -
+                                es[generate_from].df[start]).dt.days
+                    duration = duration.tolist()
+                    es['Encounter'].df['length'] = duration
+                    updated_target_entity = es['Encounter'].df
+                    duration_df = pd.DataFrame({'object_id': duration})
+
+                    es = es.entity_from_dataframe(
+                        entity_id='Duration',
+                        dataframe=duration_df,
+                        index='object_id')
+
+                    es = es.entity_from_dataframe(entity_id='Encounter',
```

Comment:
@ShahadThobaiti Any reason we don't change this to `self.target_entity`
```suggestion
                    es = es.entity_from_dataframe(entity_id=self.target_entity,
```